src/Install/DBusRaucInstaller.cpp

```diff
@@ -82,79 +159,106 @@ namespace sua {
 
     std::string DBusRaucInstaller::getBundleVersion()
     {
-        return getDBusRaucBundleVersion();
+        std::string bundleVersion = getDBusRaucBundleVersion();
+        g_main_context_iteration(g_main_loop_get_context(loop), FALSE);
+
+        if(bundleVersion == VERSION_UNAVAILABLE) {
+            bundleVersion = getOsVersionId();
+        }
+        return bundleVersion;
     }
 
     std::string DBusRaucInstaller::getBundleVersion(const std::string& input)
     {
-        return getDBusRaucBundleVersion(input);
+        std::string bundleVersion = getDBusRaucBundleVersion(input);
+        g_main_context_iteration(g_main_loop_get_context(loop), FALSE);
```

Comment:
Pending D-Bus signals from bundle version check are handled here so that less signals are logged at installation.